transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/CctBackendFactory.java

```diff
@@ -15,26 +15,35 @@
 package com.google.android.datatransport.cct;
 
 import androidx.annotation.Keep;
-import androidx.annotation.VisibleForTesting;
 import com.google.android.datatransport.runtime.backends.BackendFactory;
 import com.google.android.datatransport.runtime.backends.CreationContext;
 import com.google.android.datatransport.runtime.backends.TransportBackend;
 
 @Keep
 public class CctBackendFactory implements BackendFactory {
-  private static final String URL =
+  static final String CCT_URL =
       mergeStrings("hts/frbslgiggolai.o/0clgbth", "tp:/ieaeogn.ogepscmvc/o/ac");
 
+  static final String LFLG_URL =
+      mergeStrings("hts/frbslgigp.ogepscmv/ieo/eaylg", "tp:/ieaeogn-agolai.o/1frlglgc/o");
+
   @Override
   public TransportBackend create(CreationContext creationContext) {
+    final String url;
+    // Since legacy flg and clearcut APIs are identical, they share the same backend.
+    if (creationContext.getBackendName().equals(LegacyFlgDestination.DESTINATION_NAME)) {
+      url = LFLG_URL;
+    } else {
+      url = CCT_URL;
+    }
+
     return new CctTransportBackend(
         creationContext.getApplicationContext(),
-        URL,
+        url,
         creationContext.getWallClock(),
         creationContext.getMonotonicClock());
   }
 
-  @VisibleForTesting
   static String mergeStrings(String part1, String part2) {
```

Comment:
Since this is now used by multiple classes, consider extracting into its own class not to couple every caller to this factory class